caretaker/collector.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2016 SAP SE
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import fnmatch
+import os
+import sqlite3
+import socket
+
+from swift.account.backend import AccountBroker
+
+
+def collect(device_dir='/srv/node', stale_reads_ok=False,
+            reseller_prefix='AUTH_'):
+    matches = []
+    # Search the account DB files in the device dir
+    for root, dirnames, filenames in os.walk(device_dir):
+        if fnmatch.fnmatch(root, '*/accounts/*'):
+            for filename in fnmatch.filter(filenames, '*.db'):
+                matches.append(os.path.join(root, filename))
```

Comment:
This looks like it can be shortened into
```python
matches = glob.glob(os.path.join(root, '*/accounts/*.db'))
```
See https://docs.python.org/2/library/glob.html